### PR TITLE
Add a dynamically shaped mhlo.dot lowering example

### PR DESCRIPTION
**Pull request by silvasean**
I don't know how this table of contents page is generated (https://google.github.io/iree/ir-examples) or how to actually generate the final documentation page and preview it before pushing. Please advise.

### CODE REVIEW

File: build_tools/cmake/build_docs.sh [hunk 1]
```diff
@@ -61,63 +61,59 @@ PYTHON_BIN=`which python3` scripts/update_e2e_coverage.py ${BUILD_DIR}
...
+  sed -i '/^$/N;/^\n$/D' ${filepath}
+  echo -e '```' >> ${filepath}
+}
+
+mkdir -p ${BUILD_DIR}/doc/ir_examples
+
+build_one_example iree/samples/ops/mhlo-dot.mlir llvm-ir
+build_one_example iree/samples/ops/mhlo-dot.mlir vmla
+build_one_example iree/samples/ops/mhlo-dot.mlir vulkan-spirv
+build_one_example iree/samples/ops/dynamic-mhlo-dot.mlir vmla
```
Comment (ScottTodd) [inline build_tools/cmake/build_docs.sh:119]: Nice, this reads more clearly than the previous setup.

If the `build_one_example` function itself gets more complex, we should consider moving it into a separate (maybe Python?) file.

Comment (silvasean) [reply build_tools/cmake/build_docs.sh:119]: Agreed. I think we can use MLIR hooks to capture a more structured trace. Maybe we can generate sliders or something more dynamic showing the IR evolution over time or something.

**Comment by silvasean** [comment]
> Were you able to preview the page by building the docs locally? The table of contents should be automatically generated and I wouldn't expect issues there.

Yep, I looked at the .md page locally. Thanks for your help on discord.

**Comment by silvasean** [comment]
For the record, the index file is edited directly on the `gh-pages` branch: https://github.com/google/iree/commit/733f4dcbd66a894e46fe6263bc13d52a4a6c7018#diff-d4ad6fcf3b062e6b3dbe2a806908e3a9

And the table of contents is generated by prepare_doc_publication.py in this github workflow: https://github.com/google/iree/blob/df727ae310333b274e8230acc0928e32c2f6cb8a/.github/workflows/publish_docs.yml#L44

**Comment by GMNGeoffrey** [comment]
Just a reminder to please not use merge commits for single commits like this. It adds extra noise to the history.

**Comment by silvasean** [comment]
I just used the github UI. Can we configure the repo to "not do merge commits for trivial commits when merging with the github UI"?

**Comment by GMNGeoffrey** [comment]
> I just used the github UI. Can we configure the repo to "not do merge commits for trivial commits when merging with the github UI"?

No :disappointed: You have to select "squash and merge" or "merge" from the dropdown. We could make all merging happen with a bot that could do smart things...

**Comment by silvasean** [comment]
:( 

Is there a "github feature request" we can file here? We will want to avoid this trap for npcomp as well.

**Comment by ScottTodd** [comment]
Doesn't GitHub remember your last used merge method? For most people it should consistently use squash and merge by default (once set). For IREE build cops who need to create merge commits (or people that want merge commits for their big PRs like Ben :P), they have the extra step of remembering which to use when.

**Comment by GMNGeoffrey** [comment]
> :(
> 
> Is there a "github feature request" we can file here? We will want to avoid this trap for npcomp as well.

There are various places to put FRs. The [official one](https://support.github.com/contact) doesn't get tracked publicly, so there are also public trackers, like https://github.com/isaacs/github.

You can turn merge commits off entirely for the repo, but we need them (mostly for integrates).

But Scott is right that it should remember the last strategy you used.

**Comment by silvasean** [comment]
Sorry, just to be clear, "Squash and merge" does or does not create a merge commit (Scott's description seemed to possibly imply that using it wouldn't create a merge commit).

**Comment by GMNGeoffrey** [comment]
> Sorry, just to be clear, "Squash and merge" does or does not create a merge commit? (Scott's description seemed to possibly imply that using it wouldn't create a merge commit).

It does not

**Comment by silvasean** [comment]
Ah, ok. I'll use that in the future. Very confusing that it has "merge" in the name even though it doesn't create a merge commit.

**Pull request merged**